drop-in/src/main/java/com/adyen/checkout/dropin/ui/DropInActivity.kt

```diff
@@ -151,10 +151,15 @@ class DropInActivity : AppCompatActivity(), DropInBottomSheetDialogFragment.Prot
         }
 
         if (noDialogPresent()) {
-            if (dropInViewModel.showPreselectedStored) {
-                showPreselectedDialog()
-            } else {
-                showPaymentMethodsDialog()
+            when {
+                dropInViewModel.shouldSkipToSinglePaymentMethod() -> {
+                    val firstPaymentMethod = dropInViewModel.paymentMethodsApiResponse.paymentMethods?.get(0)
+                    if (firstPaymentMethod != null) {
+                        showComponentDialog(firstPaymentMethod)
+                    }
+                }
```

Comment:
Shouldn't this also check if there are multiple (or no) payment methods?
```suggestion
                dropInViewModel.shouldSkipToSinglePaymentMethod()
                    && dropInViewModel.paymentMethodsApiResponse.paymentMethods?.size == 1  -> {
                    val firstPaymentMethod = dropInViewModel.paymentMethodsApiResponse.paymentMethods?.get(0)
                    if (firstPaymentMethod != null) {
                        showComponentDialog(firstPaymentMethod)
                    }
                }
```